docs/modules/data-structures/pages/vector-collections.adoc

```diff
@@ -0,0 +1,495 @@
+);
+----
+--
+====
+
+=== Read entries
+To get document from a vector collection, use the method `getAsync` of the `VectorCollection` class.
+
+[tabs]
+====
+Java::
++
+--
+[source,java]
+----
+CompletionStage<VectorDocument<String>> result = vectorCollection.getAsync("1");
+----
+--
+====
+
+=== Update entries
+To update single entry in a vector collection, use the method `putAsync` or `setAsync` of the `VectorCollection` class.
+
+[tabs]
+====
+Java::
++
+--
+[source,java]
+----
+VectorDocument<String> document = VectorDocument.of("{'genre': 'fiction', 'year': 2022}", VectorValues.of(new float[] {1.2f, -0.3f, 2.2f, 0.4f, 0.3f}));
+CompletionStage<Void> result = vectorCollection.setAsync("1", document);
+----
+--
+====
+
+=== Delete entries
+To delete document from a vector collection, use the method `deleteAsync` or `removeAsync` of the `VectorCollection` class.
+
+[tabs]
+====
+Java::
++
+--
+[source,java]
+----
+CompletionStage<Void> resultDelete = vectorCollection.deleteAsync("1");
+CompletionStage<VectorDocument<String>> resultRemove = vectorCollection.removeAsync("2");
+----
+--
+====
+
+NOTE: These methods do not delete vectors but mark them as deleted. This may impact search speed and memory usage. To permanently remove vectors from the index, you must run index optimization after deletion. See <<optimize-collection, optimize method>>
+
+== Similarity search
+
+Vector search returns entries with vectors that are most similar to the query vector, based on specified metrics. Any query consists of exactly one vector to search and search options, such as the limit of results to retrieve, etc. For more details about available options, refer to <<similarity-search-options, Similarity search options>>
+
+For similarity search, use the `searchAsync` method of the `VectorCollection`.
+
+In a single index vector collection, specifying the name of the index to search is not required.
+However, for a multi-index vector collection, specifying the name of the index to search in is necessary.
+
+Example for single-index vector collection:
+[tabs]
+====
+Java::
++
+--
+[source,java]
+----
+CompletionStage<SearchResults<String, String>> results = vectorCollection.searchAsync(
+        VectorValues.of(new float[] {0f, 0f, 0.2f, -0.3f, 1.2f}),
+        SearchOptions.builder()
+            .limit(5)
+            .includeVectors()
+            .includeValue()
+            .build()
+);
+----
+--
+====
+
+Example for multi-index vector collection:
+[tabs]
+====
+Java::
++
+--
+[source,java]
+----
+CompletionStage<SearchResults<String, String>> results = vectorCollection.searchAsync(
+        VectorValues.of("glove-index", new float[] {0f, 0f, 0.2f, -0.3f, 1.2f}),
+        SearchOptions.builder()
+            .limit(5)
+            .includeVectors()
+            .includeValue()
+            .build()
+);
+----
+--
+====
+
+=== Similarity search options
+Search parameters are passed as an argument `searchOptions` to the searchAsync method.
```

Comment:
I think we should document somewhere the impact of partition count:
1. on ingestion: more give better parallelism (up to some point probably around total number of partition threads in the cluster)
2. on search: less gives better performance, impact on quality is complicated, depends on `partitionLimit` - IMO tuning `partitionLimit` is quite important in 5.5 to get reasonable results. WARNING: default 271 partitions may be inefficient for search for moderately large collections.
3. on migration: very big partitions should be avoided (current recommendation is still around 50-100MB/partiion), which gives quicker migration and smaller pressure on heap during migration (especially until chunked migration is implemented) which means that number of partitions should be adjusted to the size of data
4. on other data structures: number of partitions is a cluster-wide setting shared by all data structures. If the needs are vastly different, it might be considered to create separate clusters

Reply:
I don't think we currently have any specific recommendation regarding partition count and size (except generic 100MB size) except that it should be tuned to given collection size, use, required search latency, recall and updates

Reply:
i will add it to https://github.com/hazelcast/hz-docs/pull/1134/